src/main/java/app/linguistai/bmvp/model/gamification/IXPRanking.java

```diff
@@ -0,0 +1,23 @@
+package app.linguistai.bmvp.model.gamification;
+
+import java.nio.ByteBuffer;
+import java.util.UUID;
+
+public interface IXPRanking {
```

Comment:
As far as I can see, no class implements this interface. In that case, it would be better to make this interface a regular class. Is there a benefit of this class being an interface? Would it be possible to change this to class? Thanks

Reply:
This is a data projection interface: 
https://docs.spring.io/spring-data/jpa/reference/repositories/projections.html#projections.interfaces.closed

Native queries do not support JPA's automatic mapping with classes, instead we define a projection interface: 
https://thorben-janssen.com/spring-data-jpa-dto-native-queries/

Reply:
Thanks!